### PR TITLE
chore(greenhouse): migrate to vite

### PR DESCRIPTION
**Pull request by andypf**
# Summary

This PR migrates Greenhouse Dashboard to Vite, enhancing our build system consistency.

## Additional Notes:
These changes will streamline our build process and improve compatibility within our ecosystem.

# Changes Made

- Removed esbuild-related files.
- Added `vite.config.js` for Vite configuration.
- Introduced the `build:static` script for static builds.
- Updated the `Dockerfile` to replace `build` with `build:static`. This change ensures that Vite generates a build containing an `index.html`, rendering the previous `index.html` in the Docker folder obsolete.
- Modified worker imports in Supernova to use `?worker&inline`. This is crucial for supporting workers within the Greenhouse context. Without `inline`, Vite generates a separate chunk for each worker, which complicates the resolution of hard-coded worker URLs in Supernova. The `inline` option resolves this issue.

# Related Issues

- #429 

# Testing Instructions

1. `npm i`
2. `npm run TASK`

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

**Comment by andypf** [comment]
closes #429

**Review by ArtieReus — CHANGES_REQUESTED** [review]
checking. Need more time

File: .github/workflows/deploy-pr-preview.yaml [hunk 1]
```diff
@@ -169,15 +169,9 @@ jobs:
...
-          cp -r "$PACKAGE_PATH/$build_folder" "$DEPLOY_PATH/$TARGET_FOLDER/$build_folder"
-          cp "$PACKAGE_PATH/public/favicon.ico" "$DEPLOY_PATH/$TARGET_FOLDER/"
-          cp "$PACKAGE_PATH/public/favicon.svg" "$DEPLOY_PATH/$TARGET_FOLDER/"
-
-          # Generate index.html and appProps.json
-          cp "$PACKAGE_PATH/docker/index.html" "$DEPLOY_PATH/$TARGET_FOLDER/index.html"
-          entry_file_escaped=$(echo "$entry_file" | sed 's/\//\\\//g')
-          sed -i 's|/appProps.json?%VERSION%|./appProps.json|g' "$DEPLOY_PATH/$TARGET_FOLDER/index.html"
-          sed -i "s|/index.js?%VERSION%|./$entry_file_escaped|g" "$DEPLOY_PATH/$TARGET_FOLDER/index.html"
+          cp -r "$PACKAGE_PATH/$build_folder" "$DEPLOY_PATH/$TARGET_FOLDER"
```
Comment (ArtieReus) [inline .github/workflows/deploy-pr-preview.yaml:172]: Once this PR is merged
https://github.com/cloudoperators/juno/pull/472

just do following:
```
  ./.github/scripts/build-vite-app.sh
```

File: apps/greenhouse/.gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 public/extensions
+appProps.json
```
Comment (ArtieReus) [inline apps/greenhouse/.gitignore:2]: Do we need again appProps.json? We have already a global entry for it:
https://github.com/cloudoperators/juno/blob/main/.gitignore#L8

File: apps/greenhouse/package.json [hunk 3]
```diff
@@ -18,17 +18,12 @@
     "@svgr/core": "^7.0.0",
     "@svgr/plugin-jsx": "^7.0.0",
     "@tailwindui/react": "^0.1.1",
-    "@testing-library/dom": "^8.19.0",
     "@testing-library/jest-dom": "^5.17.0",
```
Comment (ArtieReus) [inline apps/greenhouse/package.json:21]: Do we need this libs?
```
@testing-library/jest-dom
@testing-library/react
```
I removed them from the other vitificatioins.

File: apps/greenhouse/package.json [hunk 4]
```diff
@@ -18,17 +18,12 @@
...
     "@testing-library/react": "^16.0.1",
-    "@testing-library/user-event": "^14.4.3",
     "assert": "^2.0.0",
```
Comment (ArtieReus) [inline apps/greenhouse/package.json:23]: This one `"assert": "^2.0.0"` I also removed from the other vitifications.

File: apps/greenhouse/package.json [hunk 5]
```diff
@@ -18,17 +18,12 @@
...
     "autoprefixer": "^10.4.2",
-    "babel-jest": "^29.3.1",
-    "esbuild": "^0.20.2",
     "github-markdown-css": "^5.7.0",
     "immer": "^9.0.21",
-    "jest": "^29.3.1",
     "postcss": "^8.4.47",
     "postcss-url": "^10.1.3",
```
Comment (ArtieReus) [inline apps/greenhouse/package.json:28]: This one `postcss-url` also has been removed from the other vitifications.

File: apps/greenhouse/package.json [hunk 6]
```diff
@@ -38,18 +33,20 @@
...
     "zustand": "4.5.5"
   },
   "scripts": {
     "lint": "eslint",
     "test": "vitest run",
-    "dev": "NODE_ENV=development node esbuild/config.js --serve --watch",
-    "build": "NODE_ENV=production node esbuild/config.js",
+    "dev": "node scripts/handle_extensions.js && vite",
+    "build": "node scripts/handle_extensions.js && vite build",
+    "build:static": "node scripts/handle_extensions.js && vite build --mode static",
```
Comment (ArtieReus) [inline apps/greenhouse/package.json:49]: we still need to add as we did for the other vitifications
```
"serve": "vite preview",
```

File: apps/greenhouse/package.json
Comment (ArtieReus) [inline apps/greenhouse/package.json:1]: Everything withing the `appProps ` should be migrated to the readme:
https://github.com/cloudoperators/juno/pull/499/files#diff-56db823e2d993fbde8cc3bcca2c0a6fb574ded7b4262e2e685bf8115992d06daR64

Comment (ArtieReus) [inline apps/greenhouse/package.json:1]: the key `appPreview ` can be removed

File: apps/greenhouse/vite.config.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      port: parseInt(process.env.PORT || "3000"),
+    },
+  }
+
+  // with vite it is possible to have different configurations based on the mode
+  // we can use this to create a static build for previewing the app in github pages
+  // and also to create a docker image for the standalone app
+  if (mode === "static") {
+    return {
+      ...sharedConfig,
```
Comment (ArtieReus) [inline apps/greenhouse/vite.config.ts:38]: we need this for the preview:
```
 base: "./", // Relative Path in Generated index.html
```

**Review by ArtieReus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**